modular_hearthstone/code/modules/spells/roguetown/wizard.dm

```diff
@@ -1539,7 +1539,124 @@
 		REMOVE_TRAIT(H, TRAIT_MIRROR_MAGIC, TRAIT_GENERIC)
 		to_chat(H, span_warning("Your connection to mirrors fades away."))
 
+/obj/effect/proc_holder/spell/invoked/shadowstep
+	name = "Shadowstep"
+	desc = "Project your shadow to swap places with it, teleporting several feet away."
+	cost = 1
+	xp_gain = TRUE
+	releasedrain = 30
+	warnie = "spellwarning"
+	movement_interrupt = TRUE
+	associated_skill = /datum/skill/magic/arcane
+	overlay_state = "shadowstep"
+	chargedrain = 1
+	chargetime = 0 SECONDS
+	charge_max = 30 SECONDS
+	var/area_of_effect = 1
+	var/max_range = 7
+	var/turf/destination_turf
+	var/turf/user_turf
+	var/mutable_appearance/tile_effect
+	var/leash_range = 7
+	var/mutable_appearance/target_effect
+	var/datum/looping_sound/invokeshadow/shadowloop
+
+//Resets the tile and turf effects.
+/obj/effect/proc_holder/spell/invoked/shadowstep/proc/reset(silent = FALSE)
+	if(tile_effect && destination_turf)
+		destination_turf.cut_overlay(tile_effect)
+		qdel(tile_effect)
+		destination_turf = null
+	if(user_turf && target_effect)
+		user_turf.cut_overlay(target_effect)
+		qdel(target_effect)
+		user_turf = null
+	update_icon()
+
+/obj/effect/proc_holder/spell/invoked/shadowstep/proc/check_path(turf/Tu, turf/Tt)
+	var/dist = get_dist(Tt, Tu)
+	var/last_dir
+	var/turf/last_step
+	if(Tu.z > Tt.z) 
+		last_step = get_step_multiz(Tu, DOWN)
+	else if(Tu.z < Tt.z)
+		last_step = get_step_multiz(Tu, UP)
+	else 
+		last_step = locate(Tu.x, Tu.y, Tu.z)
+	var/success = FALSE
+	for(var/i = 0, i <= dist, i++)
+		last_dir = get_dir(last_step, Tt)
+		var/turf/Tstep = get_step(last_step, last_dir)
+		if(!Tstep.density)
+			success = TRUE
+			var/list/cont = Tstep.GetAllContents(/obj/structure/roguewindow)
+			for(var/obj/structure/roguewindow/W in cont)
+				if(W.climbable && !W.opacity)	//It's climable and can be seen through
```

Comment:
clim**b**able

Reply:
you did this!!!

Reply:
> you did this!!!

and I did it WRONG. Don't repeat the WRONG.